spec/lib/task_helpers/imports/custom_buttons_spec.rb

```diff
@@ -41,32 +44,53 @@
 
       context "without existing buttons" do
         context "only imports good yaml" do
-          it 'imports a specified file' do
-            TaskHelpers::Imports::CustomButtons.new.import(options)
-            assert_test_custom_button_set_present
-            assert_imports_only_custom_button_set_one
+          context "connect dialog flag is set" do
+            it 'imports a specified file' do
+              TaskHelpers::Imports::CustomButtons.new.import(options)
+              assert_test_custom_button_set_present
+              assert_imports_only_custom_button_set_one
+              assert_dialog_is_set(true)
+            end
+          end
+          context "connect dialog flag not set" do
+            let(:connect_dialog_by_name) { false }
+            it 'imports a specified file' do
+              TaskHelpers::Imports::CustomButtons.new.import(options)
+              assert_test_custom_button_set_present
+              assert_imports_only_custom_button_set_one
+              assert_dialog_is_set(false)
+            end
           end
         end
+      end
 
-        context "doesn't import bad yaml" do
-          let(:source) { "#{data_dir}/#{bad_custom_button_file}" }
-          it 'does not imports a specified file' do
-            TaskHelpers::Imports::CustomButtons.new.import(options)
-            assert_imports_no_custom_buttons
-          end
+      context "doesn't import bad yaml" do
+        let(:source) { "#{data_dir}/#{bad_custom_button_file}" }
+        it 'does not imports a specified file' do
+          TaskHelpers::Imports::CustomButtons.new.import(options)
+          assert_imports_no_custom_buttons
         end
       end
 
       context "with existing identical buttons" do
         it 'should not import anything' do
           TaskHelpers::Imports::CustomButtons.new.import(options)
-          assert_raises_import_error
           assert_imports_only_custom_button_set_one
         end
       end
     end
   end
 
+  def assert_dialog_is_set(connect)
+    btn1 = CustomButton.find_by(:name => custom_button_1_name)
+    expect(btn1).to be
```

Comment:
This line seems not right.

Reply:
```
  button_1 = CustomButton.find_by(:name => "button 1")
  expect(button_1).not_to be_nil
  if connect
       expect(button_1.resource_action.dialog.id).to eq(test_dialog_2.id)
     else
       expect(button_1.resource_action.dialog).to be_nil
     end
  end
```

Reply:
@borisko123 Can you fix this line `expect(btn1).to be`?

Reply:
Hi @lfu , what is wrong with this line?
It asserts that button exists...
-----
from : https://relishapp.com/rspec/rspec-expectations/v/3-9/docs/built-in-matchers/be-matchers
expect(obj).to be         # passes if obj is truthy (not nil or false)

Reply:
<img width="802" alt="Screen Shot 2020-06-16 at 13 35 56" src="https://user-images.githubusercontent.com/2593270/84808227-645e8680-afd6-11ea-99a3-4e356c1efc09.png">

Reply:
@lfu Done